lib/swagger-connector.js

```diff
@@ -105,7 +105,7 @@ SwaggerConnector.prototype.connect = function(cb) {
     }
 
     var client = new SwaggerClient({
-      url: self.url,
+      url: self.settings.url,
```

Comment:
@deepakrkris : this will break. `spec` parameter is used to provide a URL where actual Swagger specification is hosted and that's where `swagger-client` reads specification from. Thus, changing `self.url` doesn't really help with adding `host` value to actual Swagger specification. [See here](https://github.com/strongloop/loopback-connector-swagger/blob/a656eedd5490869152ed3e62a708bc4de13bf00d/lib/spec-resolver.js#L23-L33) how `url` -- the URL for hosted swagger specification -  is retrieved when `settings.spec` is a `string`. 

Now, the question is: how to put `host` when it's missing in given Swagger specification. IIRC, `swagger-client` module takes care of resolving `host` -- when it's not present in specification -- to where the `http` request came from.  ( Please verify it [here](https://github.com/swagger-api/swagger-js/tree/2.x) if that's the case or not)

To take control of resolving `host`, I recommend updating `host` with the one from `settings` at [this place](https://github.com/strongloop/loopback-connector-swagger/blob/a656eedd5490869152ed3e62a708bc4de13bf00d/lib/swagger-connector.js#L92) as follows:

```javascript
if(!self.spec.host){
 self.spec.host = self.settings.apiHost // I recommend naming this something else than `url` as it may create confusion
}
```

Hope that helps.